### PR TITLE
[vendor] Add support for Kubernetes 1.11

### PR DESCRIPTION
**Pull request by surajssd**
This commit bumps up the version of all the Kubernetes dependencies
to `1.11.2`.

Which includes:

- k8s.io/client-go
- k8s.io/api
- k8s.io/apiextensions-apiserver
- k8s.io/apimachinery
- k8s.io/code-generator

Fixes https://github.com/habitat-sh/habitat-operator/issues/341

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -114,6 +114,12 @@ defaults: &defaults
         when: always
 
 jobs:
+  k8s-v1.11:
+    environment:
+      K8S_VERSION: v1.11.2
+      GKE_CLUSTER_VERSION: "1.11.0-gke.1"
```
Comment (kosyfrances) [inline .circleci/config.yml]: @surajssd this cluster version is not supported. 
```ERROR: (gcloud.container.clusters.create) ResponseError: code=400, message=Master version "1.11.0-gke.1" is unsupported.```

**Comment by surajssd** [comment]
Since GKE has not rolled out the Kubernetes 1.11 rightfully yet, so we will solve this https://github.com/habitat-sh/habitat-operator/issues/343 first and then this PR could just be add support for 1.11.

File: .circleci/config.yml [hunk 2]
```diff
@@ -107,6 +107,12 @@ defaults: &defaults
         when: always
 
 jobs:
+  k8s-v1.11:
+    environment:
+      K8S_VERSION: v1.11.2
+      GKE_CLUSTER_VERSION: "1.11.2-gke.3"
+      CODEGEN_VERSION: kubernetes-1.11.2
```
Comment (krnowak) [inline .circleci/config.yml]: This is not used anymore, please drop this line.

File: .circleci/config.yml [hunk 3]
```diff
@@ -107,6 +107,12 @@ defaults: &defaults
         when: always
 
 jobs:
+  k8s-v1.11:
+    environment:
+      K8S_VERSION: v1.11.2
+      GKE_CLUSTER_VERSION: "1.11.2-gke.3"
```
Comment (krnowak) [inline .circleci/config.yml]: Won't using just `"1.11"` work?

Comment (surajssd) [reply .circleci/config.yml]: it is not GA yet so can't use it right now!

File: .circleci/config.yml (hunk 2 above)
Comment (surajssd) [reply .circleci/config.yml]: oh crap missed it

Comment (surajssd) [reply .circleci/config.yml]: good catch, updated it!

File: .circleci/config.yml (hunk 3 above)
Comment (krnowak) [reply .circleci/config.yml]: Alright.

**Pull request merged**